src/commands/hardis/org/diagnose/unusedusers.ts

```diff
@@ -0,0 +1,171 @@
+  protected outputFile;
+  protected outputFilesRes: any = {};
+  protected lastNdays: number;
+  protected unusedUsers = [];
+  protected statusCode = 0;
+
+  /* jscpd:ignore-end */
+
+  public async run(): Promise<AnyJson> {
+    this.debugMode = this.flags.debug || false;
+    this.outputFile = this.flags.outputfile || null;
+    this.lastNdays = this.flags.days;
+    this.debugMode = this.flags.debug || false;
+    this.outputFile = this.flags.outputfile || null;
+
+    // If manual mode and days not sent as parameter, prompt user
+    if (!isCI && !this.lastNdays) {
+      const lastNdaysResponse = await prompts({
+        type: "select",
+        name: "days",
+        message: "Please select the period to detect inactive active users.",
+        choices: [
+          { title: `7 days`, value: 7 },
+          { title: `30 days`, value: 30 },
+          { title: `90 days`, value: 90 },
+          { title: `6 months (180 days)`, value: 180 },
+          { title: `1 year (365 days)`, value: 365 },
+          { title: `2 years (730 days)`, value: 730 }
+        ],
+      });
+      this.lastNdays = lastNdaysResponse.days;
+    }
+
+    const conn = this.org.getConnection();
+    uxLog(this, c.cyan(`Extracting active users who haven't logged in for a while on ${conn.instanceUrl} ...`));
+
+    // Retrieve the list of users who haven't logged in for a while
+    this.unusedUsers = await this.listUnusedUsersWithSfdcLicense(conn);
+
+    // Generate output CSV file
+    if (this.unusedUsers.length > 0) {
+      this.outputFile = await generateReportPath("unused-users", this.outputFile);
+      this.outputFilesRes = await generateCsvFile(this.unusedUsers, this.outputFile);
+    }
+
+    // Manage notifications
+    await this.manageNotifications();
+
+    const userSummaryInfo = this.unusedUsers.length == 1 ? 'user has' : 'users have'
+    let msg = `No unused users have been found`;
+    let summary;
+    if(this.unusedUsers.length == 0){
+      summary = `All users have logged in to ${conn.instanceUrl} within the last ${this.lastNdays} days!`;
+    } else {
+      this.statusCode = 1;
+      msg = `${this.unusedUsers.length} unused ${userSummaryInfo} been found`;
+      summary = `${this.unusedUsers.length} active ${userSummaryInfo} not logged in to ${conn.instanceUrl} in the last ${this.lastNdays} days!`;
+    }
+
+    if ((this.argv || []).includes("unusedusers")) {
+      process.exitCode = this.statusCode;
+    }
+
+    // Return an object to be displayed with --json
+    return {
+      status: this.statusCode,
+      message: msg,
+      summary: summary,
+      unusedUsers: this.unusedUsers,
+      csvLogFile: this.outputFile,
+      xlsxLogFile: this.outputFilesRes.xlsxFile,
+    };
+  }
+
+  private async listUnusedUsersWithSfdcLicense(conn) {
+    const whereConstraint = `WHERE IsActive = true AND LastLoginDate != LAST_N_DAYS:${this.lastNdays} AND LastLoginDate != NULL AND Profile.UserLicense.LicenseDefinitionKey = 'SFDC' `;
```

Comment:
I think you can add more licenses types by default, to handle all the "paying" ones

For example Platform & Partner Community licenses ^^

https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_userlicense.htm

Maybe add a parameter licensetypes that could be:
- all-crm that would be converted into SFDC,AUL,AUL1,AULL_IGHT... (default value)
- all-paying that would be all-crm + community types licenses
- all ( no filter on license type)
- a list of licence type Ids

+ prompted if not specified and not in CI